winforms/src/toga_winforms/widgets/detailedlist.py

```diff
@@ -1,135 +1,55 @@
-import System.Windows.Forms as WinForms
-from travertino.size import at_least
+from functools import lru_cache
 
-from .base import Widget
+from toga.sources import Row
 
+from .table import Table
+
+
+class DetailedList(Table):
+    # The following methods are overridden from Table.
+    @property
+    def _headings(self):
+        return None
+
+    @property
+    def _accessors(self):
+        return ("title", "subtitle")
+
+    @property
+    def _multiple_select(self):
+        return False
+
+    @property
+    @lru_cache
+    def _data(self):
+        interface = self.interface
+
+        class TableData:
```

Comment:
Any reason this is inline, rather than top level?

Reply:
In this case the class is only used within one function, so I was included to keep it there. But then I realized that the `lru_cache` was attached to the method object rather than any particular instance of the class, so it would keep the `interface` alive indefinitely. Moving the class to the module level was the simplest way to fix this.